dev_guide/expanding_persistent_volumes.adoc

```diff
@@ -0,0 +1,79 @@
+
+*  Watches for PVC update requests and adds the PVC to the controller's work queue if an increase in volume size was requested. Once the PVC is added to the controller's work queue, `- pvc.Status.Conditions` is updated to `ResizeStarted: True`.
+* For unbound or pending PVCs,` - resize ` triggers no action in `volume_expand_controller`.
+* If `pv.Spec.Capacity already` is greater or equal than the requested size, then no action is performed by the controller.
+* A separate goroutine reads the work queue and performs a corresponding volume resize operation. If there is a resize operation in progress for the same volume, then the resize request is pending and retries once the previous resize request is complete.
+* Controller resizing is level-based rather than edge-based. If there is more than one pending resize request for the same PVC, then new resize requests for the same PVC replace older pending requests.
+* Resize is performed via the volume plugin interface, which is executed inside a goroutine spawned by `operation_exectutor`.
+* A new plugin interface called `volume.Expander` is added to the volume plugin interface. The expander interface defines if the volume requires a file system resize.
+* The controller calls to expand the PVC.
+* Once the volume expansion is successful, the volume is marked as expanded and the new size is reflected or updated in `pv.spec.capacity`. Any errors are reported as events on the PVC object.
+* If the resize operation fails, then `pvc.Status.Conditions` is updated with `ResizeFailed: True`. The corresponding error is also added to the condition field.
+* Depending on the volume type:
+** If the volume is of a type that does not require a file system resize, then `pvc.status.capacity` is immediately updated to reflect the new size. This would conclude the volume expand operation, and  `pvc.Status.Conditions` is updated to `Ready: True`.
+** If the volume is of a type that requires a file system resize, then a file system resize is performed on kubelet. 
+* If the volume plugin is of type that cannot do resizing of attached volumes (such as Cinder), then `ExpandVolumeDevice` can return an error by checking for volume status with its own API (such as by making an Openstack Cinder API call in this case). The controller continues trying to resize the volume until it is successful.
+* To consider cases of missed PVC update events, an additional loop reconciles bound PVCs with PVs. This additional loop loops through all PVCs, matches `pvc.spec.resources.requests` with `pv.spec.capacity`, and adds the PVC in the volume_expand_controller's work queue if `pv.spec.capacity` is less than `pvc.spec.resources.requests`.
+* There are additional checks in the controller that expand the PV size. These checks ensure that you do not make volume plugin API calls that can reduce the size of the PV.
```

Comment:
plug-in

Reply:
more shop talk, mention of API or how the controller works should be avoided.